src/coreComponents/mesh/generators/WellGeneratorBase.cpp

```diff
@@ -528,54 +529,49 @@ void WellGeneratorBase::debugWellGeometry() const
     return;
   }
 
-  std::cout << std::endl;
-  std::cout << "++++++++++++++++++++++++++" << std::endl;
-  std::cout << "WellGeneratorBase = " << getName() << std::endl;
-  std::cout << "MPI rank = " << MpiWrapper::commRank( MPI_COMM_GEOSX ) << std::endl << std::endl;
-  std::cout << "Number of well elements = " << m_numElems << std::endl;
+  std::vector< string > row;
+  std::ostringstream oss;
+  Table table = Table( {
+      Table::ColumnParam{{"Element no."}, Table::Alignment::right},
+      Table::ColumnParam{{"CoordX"}, Table::Alignment::middle},
+      Table::ColumnParam{{"CoordY"}, Table::Alignment::middle},
+      Table::ColumnParam{{"CoordZ"}, Table::Alignment::middle},
+      Table::ColumnParam{{"Prev\nElement"}, Table::Alignment::right},
+      Table::ColumnParam{{"Next\nElement"}, Table::Alignment::right},
+    } );
+
+  string titleName = "InternalWellGenerator " + getName();
+  table.setTitle( titleName );
 
   for( globalIndex iwelem = 0; iwelem < m_numElems; ++iwelem )
   {
-    std::cout << "Well element #" << iwelem << std::endl;
-    std::cout << "Coordinates of the element center: " << m_elemCenterCoords[iwelem] << std::endl;
-    if( m_nextElemId[iwelem] < 0 )
-    {
-      std::cout << "No next well element" << std::endl;
-    }
-    else
-    {
-      std::cout << "Next well element # = " << m_nextElemId[iwelem] << std::endl;
-    }
-    if( m_prevElemId[iwelem][0] < 0 )
-    {
-      std::cout << "No previous well element" << std::endl;
-    }
-    else
+    std::optional< globalIndex > nextElement;
+    std::optional< globalIndex > prevElement;
```

Comment:
What is the interest of using a `std::optional` here?

Reply:
There can be empty cell in the table, so i decided to not display any data : 
``` 
+--------------------------------------------------------------------------------------+
|                         InternalWellGenerator well_injector1                         |
+---------------+----------+----------+----------------------+------------+------------+
|  Element no.  |  CoordX  |  CoordY  |        CoordZ        |      Prev  |      Next  |
|               |          |          |                      |   Element  |   Element  |
+---------------+----------+----------+----------------------+------------+------------+
|            0  |   9.5    |   0.2    |  11.958333333333334  |         1  |            |
+--------------------------------------------------------------------------------------+
```

Reply:
Maybe I'm wrong, but your `nextElement` and `prevElement` are always non empty. So why don't you define them as
```c++
globalIndex nextElement;
globalIndex prevElement;
```
?

Reply:
I think there is a misunderstanding.
In the case of internal well generator table, the first `nextElement` and the last `prevElement` contain no data, that's why i decided to set `next element` and `prevElement` to `std:optional` in order to display them correctly
I've added also in StringUtilities.hpp an overloading operator `<<` for std:optiona